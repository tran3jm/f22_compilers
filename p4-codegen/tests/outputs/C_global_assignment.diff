--- outputs/C_global_assignment.txt	2022-11-09 15:29:41.041619033 -0500
+++ expected/C_global_assignment.txt	2022-11-08 11:52:50.535967504 -0500
@@ -2,11 +2,11 @@
   push BP
   i2i SP => BP
   addI SP, 0 => SP
-  loadI 10 => r2
-  loadI 256 => r3
-  storeAI r2 => [r3+0]
-  loadI 0 => r4
-  i2i r4 => RET
+  loadI 10 => r0
+  loadI 256 => r1
+  storeAI r0 => [r1+0]
+  loadI 0 => r2
+  i2i r2 => RET
   jump l0
 l0:
   i2i BP => SP
