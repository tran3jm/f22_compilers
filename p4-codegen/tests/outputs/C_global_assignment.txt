main:
  push BP
  i2i SP => BP
  addI SP, 0 => SP
  loadI 10 => r2
  loadI 256 => r3
  storeAI r2 => [r3+0]
  loadI 0 => r4
  i2i r4 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return
RETURN VALUE = 0
