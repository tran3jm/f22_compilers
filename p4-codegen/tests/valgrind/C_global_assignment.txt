==2079989== Memcheck, a memory error detector
==2079989== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==2079989== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==2079989== Command: ../decaf inputs/global_assignment.decaf
==2079989== 
main:
  push BP
  i2i SP => BP
  addI SP, 0 => SP
  loadI 10 => r2
  loadI 256 => r3
  storeAI r2 => [r3+0]
  loadI 0 => r4
  i2i r4 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return
RETURN VALUE = 0
==2079989== 
==2079989== HEAP SUMMARY:
==2079989==     in use at exit: 0 bytes in 0 blocks
==2079989==   total heap usage: 1,884 allocs, 1,884 frees, 393,540 bytes allocated
==2079989== 
==2079989== All heap blocks were freed -- no leaks are possible
==2079989== 
==2079989== For lists of detected and suppressed errors, rerun with: -s
==2079989== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
