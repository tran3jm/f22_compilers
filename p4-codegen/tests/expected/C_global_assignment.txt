main:
  push BP
  i2i SP => BP
  addI SP, 0 => SP
  loadI 10 => r0
  loadI 256 => r1
  storeAI r0 => [r1+0]
  loadI 0 => r2
  i2i r2 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return
RETURN VALUE = 0
